.github/workflows/ci/pkg/environment/environment.go

```diff
@@ -111,22 +118,82 @@ func New(c Config) (*PullRequestEnvironment, error) {
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
+	// Check if the pull request has changes to the `docs/` directory as that will effect who
+	// the required reviewers are.
+	docChanges, codeChanges, err := hasChanges(c.Context, pr, c.Client)
+	if err != nil {
+		// Log the error, don't fail the whole run because it couldn't detect if the pull request has docs
+		log.Errorf("error while detecting pull request for docs changes: %v, skipping assigning docs reviewers", err)
+	}
+
 	return &PullRequestEnvironment{
 		Client:           c.Client,
 		reviewers:        c.Reviewers,
 		defaultReviewers: c.Reviewers[ci.AnyAuthor],
 		Metadata:         pr,
+		HasDocsChanges:   docChanges,
+		HasCodeChanges:   codeChanges,
 	}, nil
 }
 
+// hasChanges determines if the pull request has changes to `docs/`
+func hasChanges(ctx context.Context, pr *Metadata, clt *github.Client) (hasDocsChanges bool, hasCodeChanges bool, err error) {
+	files, err := getPullRequestFiles(ctx, pr, clt)
+	if err != nil {
+		return false, true, trace.Wrap(err)
+	}
+	for _, file := range files {
+		if file.Filename == nil {
+			return false, true, trace.BadParameter("pull request file name is nil")
+		}
+		if strings.HasPrefix(*file.Filename, ci.DocsPrefix) {
+			hasDocsChanges = true
+		} else {
+			hasCodeChanges = true
+		}
+
+	}
+	return hasDocsChanges, hasCodeChanges, nil
+}
+
+// getPullRequestFiles gets all the files in the pull request.
+func getPullRequestFiles(ctx context.Context, pr *Metadata, clt *github.Client) ([]*github.CommitFile, error) {
+	// Can only request 100 files per page according to the docs.
+	// https://docs.github.com/en/free-pro-team@latest/rest/reference/pulls/#list-pull-requests-files
+	opt := &github.ListOptions{PerPage: 100}
+
+	var allFiles []*github.CommitFile
+	for {
+		files, resp, err := clt.PullRequests.ListFiles(ctx, pr.RepoOwner, pr.RepoName, pr.Number, opt)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		allFiles = append(allFiles, files...)
+		if resp.NextPage == 0 {
+			break
+		}
+		opt.Page = resp.NextPage
+	}
+	return allFiles, nil
+}
+
 // GetReviewersForAuthor gets the required reviewers for the current user.
 func (e *PullRequestEnvironment) GetReviewersForAuthor(user string) []string {
-	value, ok := e.reviewers[user]
-	// Author is external or does not have set reviewers
+	var reviewers []string
+	requiredReviewers, ok := e.reviewers[user]
 	if !ok {
-		return e.defaultReviewers
+		reviewers = e.defaultReviewers
+	} else {
+		reviewers = requiredReviewers
+	}
+
+	switch {
```

Comment:
Why the switch here? Wouldn't the following be simpler?

```
var reviewers []string
if e.HasDocsChanges {
    reviewers = append(reviewers, ci.DocReviewers...)
}
if e.HasCodeChanges {
    rr, ok := e.reviewers[user]
    if ok {
        reviewers = append(reviewers, rr...)
    } else {
        reviewers = append(reviewers, e.defaultReviewers...)
    }
}
```

Reply:
We still need to assign reviewers to pull requests with no code changes, just not bother the docs folks about it.